### PR TITLE
Pass branch option into recursive call within Install - for the cases whenever install is invoked with URL(s)

### LINKED ISSUE #7461: datalad install does not respect --branch option
### What is the problem?

`datalad install` seems to always checkout the default branch, ignoring the value of `--branch`

### What steps will reproduce the problem?

```shell
datalad create hello-world
cd hello-world
datalad install -d . --branch on-a-branch https://github.com/jennydaman/example-datalad.git
cd example-datalad
git branch
```

Expected: `on-a-branch`
Actual: `master`

### DataLad information

```
datalad --version
datalad 0.19.2
```

```
datalad wtf
# WTF
## configuration <SENSITIVE, report disabled by configuration>
## credentials
  - keyring:
    - active_backends:
      - SecretService Keyring
      - PlaintextKeyring with no encyption v.1.0 at /home/jenni/.local/share/python_keyring/keyring_pass.cfg
    - config_file: /home/jenni/.config/python_keyring/keyringrc.cfg
    - data_root: /home/jenni/.local/share/python_keyring
## datalad
  - version: 0.19.2
## dataset
  - branches:
    - git-annex@422291a
    - master@6488e12
  - id: d420a1f0-a32d-4243-b867-c110de28d407
  - path: /home/jenni/msGithub/another-example-datalad
  - repo: AnnexRepo
## dependencies
  - annexremote: 1.6.0
  - boto: 2.49.0
  - cmd:7z: 16.02
  - cmd:annex: 10.20230626-g801c4b7
  - cmd:bundled-git: 2.41.0
  - cmd:git: 2.41.0
  - cmd:ssh: 9.3p2
  - cmd:system-git: 2.41.0
  - cmd:system-ssh: 9.3p2
  - humanize: 4.6.0
  - iso8601: 1.1.0
  - keyring: 24.2.0
  - keyrings.alt: 4.2.0
  - msgpack: 1.0.5
  - platformdirs: 3.9.1
  - requests: 2.31.0
## environment
  - LANG: en_US.UTF-8
  - LC_MESSAGES:
  - PATH: /home/jenni/opt/bin:/home/jenni/opt/CIVET/quarantines/Linux-x86_64/CIVET-2.1.0/progs:/home/jenni/opt/CIVET/quarantines/Linux-x86_64/bin:/home/jenni/opt/itksnap-4.0.0-20230220-Linux-gcc64/bin:/home/jenni/bin:/home/jenni/.local/bin:/home/jenni/.cargo/bin:/usr/lib/ccache/bin:/home/jenni/micromamba/condabin:/usr/local/sbin:/usr/local/bin:/usr/bin:/home/jenni/.local/share/gem/ruby/3.0.0/bin:/home/jenni/.yarn/bin:/home/jenni/.conda/envs/datalad/bin:/opt/cuda/bin:/opt/cuda/nsight_compute:/opt/cuda/nsight_systems/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin
## extensions
## git-annex
  - build flags:
    - Assistant
    - Webapp
    - Pairing
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Benchmark
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions:
    - aws-0.22
    - bloomfilter-2.0.1.0
    - cryptonite-0.29
    - DAV-1.3.4
    - feed-1.3.2.0
    - ghc-8.10.7
    - http-client-0.7.9
    - persistent-sqlite-2.13.0.3
    - torrent-10000.1.1
    - uuid-1.3.15
    - yesod-1.6.1.2
  - key/value backends:
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - local repository version: 10
  - operating system: linux x86_64
  - remote types:
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions:
    - 8
    - 9
    - 10
  - upgrade supported from repository versions:
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
    - 8
    - 9
    - 10
  - version: 10.20230626-g801c4b7
## location
  - path: /home/jenni/msGithub/another-example-datalad
  - type: dataset
## metadata.extractors
## metadata.filters
## metadata.indexers
## python
  - implementation: CPython
  - version: 3.10.12
## system
  - distribution: arch
  - encoding:
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - filesystem:
    - CWD:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /home/jenni/msGithub/another-example-datalad
      - type: btrfs
    - HOME:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /home/jenni
      - type: btrfs
    - TMP:
      - max_pathlength: 4096
      - mount_opts: rw,relatime,compress=zstd:3,ssd,space_cache=v2,subvolid=263,subvol=/archlinux-root
      - path: /tmp
      - type: btrfs
  - max_path_length: 300
  - name: Linux
  - release: 6.4.4-arch1-1
  - type: posix
  - version: #1 SMP PREEMPT_DYNAMIC Wed, 19 Jul 2023 19:19:38 +0000
```

### Additional context

_No response_

### Have you had any success using DataLad before?

yes

### PR DESCRIPTION
**Pull request by yarikoptic**
Happens whenever install is invoked with a URL and --branch option. Closes #7461.

TODO:

- [x] test that case, so far tests had explicit  "source" specified and thus did not trigger this code path

### CODE REVIEW

File: datalad/distribution/install.py [hunk 1]
```diff
@@ -200,7 +200,6 @@ def __call__(
             reckless=None,
             jobs="auto",
             branch=None):
-
```
Comment (yarikoptic) [inline datalad/distribution/install.py:203]: unrelated change, I will drop

**Comment by yarikoptic** [comment]
eh, travis is unhappy but seems unrelated e.g.
```
FAILED ../datalad/customremotes/tests/test_archives.py::test_basic_scenario - datalad.runner.exception.CommandError: CommandError: 'git -c diff.ignoreSubmodules=none -c core.quotepath=false annex get -c annex.retry=3 --json --json-error-messages --json-progress -c annex.dotfiles=true -- ' |;&%b5{}'"'"'"<>ΔЙקم๗あ .dbtc '' failed with exitcode 1 under /tmp/datalad_temp_tree_test_basic_scenario14h3468n [info keys: stdout_json]
> from datalad-archives...
Unable to access these remotes: datalad-archives
  Failed to fetch any archive containing MD5E-s3--202cb962ac59075b964b07152d234b70. Tried: ['MD5E-s142--e1f8dde8c4d4160878d6a0210869cfd6.tar.gz']
  Failed to fetch any archive containing MD5E-s3--202cb962ac59075b964b07152d234b70. Tried: ['MD5E-s142--e1f8dde8c4d4160878d6a0210869cfd6.tar.gz', 'MD5E-s142--e1f8dde8c4d4160878d6a0210869cfd6.tar.gz']
  Failed to fetch any archive containing MD5E-s3--202cb962ac59075b964b07152d234b70. Tried: ['MD5E-s142--e1f8dde8c4d4160878d6a0210869cfd6.tar.gz', 'MD5E-s142--e1f8dde8c4d4160878d6a0210869cfd6.tar.gz'] [err: 'get: 1 failed']
FAILED ../datalad/customremotes/tests/test_archives.py::test_annex_get_from_subdir - datalad.runner.exception.CommandError: CommandError: 'git annex drop -- ' |;&%b5{}'"'"'"<>ΔЙקم๗あ .datc '' failed with exitcode 1
FAILED ../datalad/tests/test_archives.py::test_ExtractedArchive - AssertionError: /tmp/datalad_temp_1e1e0ce8ee_5_370fyp/ |;&%b5{}'"<>ΔЙקم๗あ .dbtc / |;&%b5{}'"<>ΔЙקم๗あ .datc  must exist
FAILED ../datalad/tests/test_archives.py::test_decompress_file[None] - assert False
FAILED ../datalad/tests/test_archives.py::test_decompress_file[strip] - assert False
= 5 failed, 1134 passed, 52 skipped, 3 xfailed, 22 warnings in 942.58s (0:15:42) =
```
so I will take it out of draft.

**Review by mslw — APPROVED** [review]
Looks like a fix to me :100:

**Comment by yarikoptic-gitmate** [comment]
PR released in [`0.19.3`](https://github.com/datalad/datalad/releases/tag/0.19.3)

**Pull request merged**